gpt_index/callbacks/llama_debug.py

```diff
@@ -0,0 +1,121 @@
+from collections import defaultdict
+from datetime import datetime
+from typing import Dict, List, Any, Optional
+
+from gpt_index.callbacks.base import BaseCallbackHandler
+from gpt_index.callbacks.schema import (
+    CBEvent,
+    CBEventType,
+    EventStats,
+    TIMESTAMP_FORMAT,
+)
+
+
+class LlamaDebugHandler(BaseCallbackHandler):
+    """Callback handler that keeps track of debug info.
+
+    This handler simply keeps track of event starts/ends, separated by event types.
+    You can use this callback handler to keep track of and debug events.
+    """
+
+    def __init__(
+        self,
+        event_starts_to_ignore: Optional[List[CBEventType]] = None,
+        event_ends_to_ignore: Optional[List[CBEventType]] = None,
+    ) -> None:
+        """Initialize the llama debug handler."""
+        self.events: Dict[CBEventType, List[CBEvent]] = defaultdict(list)

```

Comment:
very nit: use e.g. `self._events` to indicate private variable, unless you intend this to be used outside of the class.